src/components/TimeSlots.tsx

```diff
@@ -2,41 +2,226 @@ import React, { useCallback } from 'react';
 import TimeSlot from './TimeSlot';
 import { StyleSheet, Text, View } from 'react-native';
 import { theme } from '../utils/theme';
+import { IAppointment, IAvailableDates } from '../interfaces/app.interface';
 
 interface Props {
+  selectedDay: string;
   slotTimes: string[];
-  selectedTime: string;
-  setSelectedTime: (value: string) => void;
+  scheduledAppointments: IAppointment[] | undefined;
+  availableDates: IAvailableDates[];
+  setScheduledAppointments: (value: IAppointment[]) => void;
+  multipleSelection: boolean;
+  multipleSelectionStrategy:
+    | 'consecutive'
+    | 'same-day-consecutive'
+    | 'non-consecutive';
   title?: string;
   backgroundColor?: string;
   mainColor?: string;
 }
 
 const TimeSlots = ({
+  selectedDay,
   slotTimes,
-  selectedTime,
-  setSelectedTime,
+  scheduledAppointments,
+  availableDates,
+  setScheduledAppointments,
+  multipleSelection,
+  multipleSelectionStrategy,
   title = 'Select time',
   backgroundColor = theme.colors.white,
 }: Props) => {
   const onPress = (value: string) => {
-    setSelectedTime(value);
+    if (!scheduledAppointments || scheduledAppointments.length === 0)
```

Comment:
This needs to be refactored in a utility function is different file. The multiple if-else statements are not helping to understand the logic. Can you please add comments or refactor the code to be more `development` friendly?

Reply:
> This needs to be refactored in a utility function is different file. The multiple if-else statements are not helping to understand the logic. Can you please add comments or refactor the code to be more `development` friendly?

Added comments that I think explain well what's going on. I agree with the util functions file but didn't have the time to do it :disappointed: